plugins/utils/navbuddy.nix

```
@@ -0,0 +1,278 @@
+          Operator = "󰆕 ";
+          TypeParameter = "󰊄 ";
+        };
+
+      useDefaultMapping = helpers.defaultNullOpts.mkBool true ''
+        If set to false, only mappings set by user are set. Else default mappings are used for keys that are not set by user
+      '';
+
+      keymapsSilent = mkOption {
+        type = types.bool;
+        description = "Whether navbuddy keymaps should be silent";
+        default = false;
+      };
+
+      mappings =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf
+            (
+              either
+              str
+              helpers.rawType
+            )
+        )
+        ''
+          {
+            "<esc>" = "close";
+            "q" = "close";
+            "j" = "next_sibling";
+            "k" = "previous_sibling";
+
+            "h" = "parent";
+            "l" = "children";
+            "0" = "root";
+
+            "v" = "visual_name";
+            "V" = "visual_scope";
+
+            "y" = "yank_name";
+            "Y" = "yank_scope";
+
+            "i" = "insert_name";
+            "I" = "insert_scope";
+
+            "a" = "append_name";
+            "A" = "append_scope";
+
+            "r" = "rename";
+
+            "d" = "delete";
+
+            "f" = "fold_create";
+            "F" = "fold_delete";
+
+            "c" = "comment";
+
+            "<enter>" = "select";
+            "o" = "select";
+            "J" = "move_down";
+            "K" = "move_up";
+
+            "s" = "toggle_preview";
+
+            "<C-v>" = "vsplit";
+            "<C-s>" = "hsplit";
+          }
+        ''
+        ''
+           Actions to be triggered for specified keybindings. It can take either action name i.e `toggle_preview`
+          Or it can a `rawType`.
+        '';
+
+      lsp = {
+        autoAttach = helpers.defaultNullOpts.mkBool false ''
+          If set to true, you don't need to manually use attach function
+        '';
+
+        preference = helpers.mkNullOrOption (with types; listOf str) ''
+          list of lsp server names in order of preference
+        '';
+      };
+
+      sourceBuffer = {
+        followNode = helpers.defaultNullOpts.mkBool true "Keep the current node in focus on the source buffer";
+        highlight = helpers.defaultNullOpts.mkBool true "Highlight the currently focused node";
+        reorient = helpers.defaultNullOpts.mkEnum ["smart" "top" "mid" "none"] "smart" ''
+          Right section can show previews too.
+                Options: "leaf", "always" or "never"
+        '';
+        scrolloff = helpers.defaultNullOpts.mkInt null ''
+          scrolloff value when navbuddy is open
+        '';
+      };
+    };
+
+  config = let
+    setupOptions = with cfg;
+      {
+        inherit window;
+        node_markers = with nodeMarkers; {
+          inherit enabled;
+          icons = with icons; {
+            inherit leaf branch;
+            leaf_selected = nodeMarkers.icons.leafSelected;
```

```suggestion
            leaf_selected = leafSelected;
```
This should work (as you are in a scope with `with icons;`).

---

ahh yes much cleaner!!!